### PR TITLE
grpc-sys: Upgrade bindgen to the latest version

### PR DESCRIPTION
**Pull request by heftig**
As requested in https://github.com/tikv/grpc-rs/pull/452#issuecomment-729693080.

### CODE REVIEW

**Comment by BusyJay** [comment]
Thanks @heftig!

**Comment by BusyJay** [comment]
I tried to compile and test the project on the new Apple M1, this patch seemed required to get it compile.

File: grpc-sys/Cargo.toml [hunk 1]
```diff
@@ -65,5 +65,5 @@ cmake = "0.1"
 pkg-config = "0.3"
 walkdir = "2.2.9"
 # Because of rust-lang/cargo#5237, bindgen should not be upgraded util a minor or major release.
-bindgen = { version = "0.51.0", default-features = false }
+bindgen = { version = "0.55.1", default-features = false, features = ["runtime"] }
```
Comment (BusyJay) [inline grpc-sys/Cargo.toml:68]: How about updating it to 0.56?

File: grpc-sys/build.rs [hunk 2]
```diff
@@ -298,6 +298,8 @@ fn bindgen_grpc(mut config: bindgen::Builder, file_path: &PathBuf) {
         .clang_arg("-std=c++11")
         .rustfmt_bindings(true)
         .impl_debug(true)
+        .size_t_is_usize(true)
+        .disable_header_comment()
```
Comment (BusyJay) [inline grpc-sys/build.rs:302]: I suggest to keep it so people may not try to edit the generated file by hand.

Comment (heftig) [reply grpc-sys/build.rs:302]: The header comment now contains the bindgen version number, making the output less stable.

Comment (BusyJay) [reply grpc-sys/build.rs:302]: Get it.

**Review by BusyJay — APPROVED** [review]

**Review by hunterlxt — APPROVED** [review]

**Pull request merged**